component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -451,6 +451,67 @@ public static int getWindowSize(AuthenticationContext context) {
         }
     }
 
+    /**
+     *
+     * @param tenantDomain Tenant Domain
+     * @return Window size of the totp configuration.
+     * @throws AuthenticationFailedException
+     */
+    public static int getWindowSize(String tenantDomain) throws AuthenticationFailedException {
+
+        int windowSize;
+        if (TOTPAuthenticatorConstants.SUPER_TENANT_DOMAIN.equals(tenantDomain)) {
+            windowSize = Integer.parseInt(getTOTPParameters().get(TOTPAuthenticatorConstants.WINDOW_SIZE));
+        } else {
+            try {
+                windowSize = getWindowSizeFromRegistry(tenantDomain, null);
+                if (windowSize == -1) {
+                    windowSize = Integer.parseInt(
+                            IdentityHelperUtil.getAuthenticatorParameters(TOTPAuthenticatorConstants.AUTHENTICATOR_NAME)
+                                    .get(TOTPAuthenticatorConstants.WINDOW_SIZE));
+                }
+            } catch (TOTPException e) {
+                throw new AuthenticationFailedException("Cannot find the property value for windowSize", e);
+            }
+        }
+        return windowSize;
+    }
+
+    /**
+     * Get stored window size.
+     *
+     * @param tenantDomain Tenant domain name.
+     * @return Window size.
+     * @throws TOTPException On Error while getting value for window size from registry.
+     */
+    public static int getWindowSizeFromRegistry(String tenantDomain, AuthenticationContext context)
+            throws TOTPException {
+
+        Integer windowSize = null;
+        int tenantID = IdentityTenantUtil.getTenantId(tenantDomain);
+        try {
+            NodeList authConfigList = getAuthenticationConfigNodeList(tenantDomain, tenantID);
+            windowSize = Integer.parseInt(getAttributeFromRegistry(authConfigList,
+                    TOTPAuthenticatorConstants.WINDOW_SIZE));
+        } catch (RegistryException e) {
+            if (context != null) {
+                context.setProperty(TOTPAuthenticatorConstants.GET_PROPERTY_FROM_IDENTITY_CONFIG,
```

Comment:
won't this return an NPE at the end of the method while returning the `windowSize`?

Reply:
Fix with 8d29c5492906ac59c0e3b8af9a4d7b13cbc77afa